authors/apps/comments/views.py

```diff
@@ -0,0 +1,150 @@
+from .renderers import CommentRenderer, ReplyRenderer
+from django.shortcuts import get_object_or_404
+from rest_framework import generics, status
+from rest_framework.permissions import IsAuthenticatedOrReadOnly
+from .serializers import CommentSerializer, ReplySerializer
+from ..profiles.models import Profile
+from .models import Comment, Reply
+from ..articles.models import Article
+from rest_framework.response import Response
+
+
+class CommentsAPIView(generics.GenericAPIView):
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+    serializer_class = CommentSerializer
+    renderer_classes = (CommentRenderer,)
+
+    def get(self, request, slug):
+        article = get_object_or_404(Article, slug=slug)
+        comments = Comment.objects.filter(article=article)
+        serialize_data = self.serializer_class(comments, many=True)
+        return Response({"comments": serialize_data.data}, status=status.HTTP_200_OK)
+
+    def post(self, request, slug):
+        article = get_object_or_404(Article, slug=slug)
+        data = request.data
+        author = Profile.objects.get(user=request.user)
+        serializer = self.serializer_class(data=data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save(
+            author=author,
+            article=article
+        )
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+
+class CommentDetailsAPIView(generics.GenericAPIView):
```

Comment:
@sebbss, be specific with the `generic view you trying to use` For example for `post` and `get` , you can use ListCreateAPIView.
More details on https://www.django-rest-framework.org/api-guide/generic-views